### DISCUSSION TITLE
Translation platform

### DISCUSSION THREAD

**Discussion by Justman100** [forwardemail/forwardemail.net#301] [Ideas]
As far as I can see, are translations managed manually?

Would use a translation platform for the translations! Have already set one up [here](https://translate.justman10000.de)! If it were to be used officially, it would run under `translate.forwardemail.net`

Besides the possibility to host the application on my site, I could also set it up on ForwardeMail's systems!

I would also need write access to the repository to push the translations and make optimizations to the source strings

**Comment by Justman100** [comment]
@shaunwarman

**Comment by shaunwarman** [comment]
Very cool! 

> As far as I can see, are translations managed manually?

Currently, forward email mainly leverages the following 2 packages: [mandarin](https://github.com/ladjs/mandarin) and [i18n](https://github.com/ladjs/i18n). Under the covers, it currently uses google translate service, but always opportunity for improvement.

**Reply by Justman100** [reply]
I already feared that! Incidentally, it is completely stupid to split strings because of untranslatable code snippets, blocks or the like!

It's extremely confusing! Not only for translation engines!

One would have to revise it! It is also contra-productive to use the original source texts as JSON keys

**Reply by titanism** [reply]
If you have a better suggestion than to bash something, please submit a PR or new approach.  Review the source code for mandarin.
